math/mathmore/src/complex_quartic.h

```diff
@@ -149,13 +152,7 @@ gsl_poly_complex_solve_quartic (double a, double b, double c, double d,
         double R2 = R * R;
 
         disc = R2 - Q3;
-
-//       more numerical problems with this calculation of disc
-//LM - use this since it fixes case reported in issue #6900 when disc is approx 0
-//      (needs to add additional test cases where disc is ~ 0)
-        double CR2 = 729 * rcub * rcub;
-        double CQ3 = 2916 * qcub * qcub * qcub;
-        disc = (CR2 - CQ3) / 2125764.0;
+        disc_tol = ZeroTolerance * (fabs(R2) + fabs(Q3));
```

Comment:
change here to:
```
std::max(ZeroTolerance, ZeroTolerance * (fabs(R2) + fabs(Q3)));
```

Reply:
I'm not sure I agree: This seems to defy the purpose for introducing the relative tolerance; if `R2` and `Q3` are small (ie smaller than 1), the maximum will be `ZeroTolerance` and we haven't actually gained anything...

Reply:
Yes when they are small, but if they are much larger than one, then their difference is larger than ZeroTolerance, but still they are equal within numerical precision

Reply:
Ah ok, that's a different argument than you made in https://github.com/root-project/root/pull/14291#discussion_r1434553855...

Reply:
I think in the case the 2 values are very small, absolute comparison is fine, since the absolute  precision of these two values is probably at the level of double precision epsilon (~1.E-15) since they are coming from a computation involving subtractions of floating point numbers. When they are large maybe this is not the case, so I think it is safer in this case to use relative comparison. 
If you can make this change, I think we can then merge the PR and hopefully fixing the failing test. Thanks

Reply:
Hi @lmoneta I already did in the last push. I'll wait to see the (hopefully green) CI results.